libcron/include/libcron/Cron.h

```diff
@@ -4,23 +4,44 @@
 #include <chrono>
 #include <queue>
 #include <memory>
+#include <mutex>
 #include "Task.h"
 #include "CronClock.h"
 
 namespace libcron
 {
-    template<typename ClockType>
+    class TaskLockerNone 
+    {
+        public:
+            void lock() {}
+            void unlock() {}
+    };
+
+    class TaskLocker
+    {
+        public:
+            TaskLocker() : lck(m, std::defer_lock) {}
+            void lock() { lck.lock(); }
+            void unlock() { lck.unlock(); }
+        private:
+            std::mutex m{};
+            std::unique_lock<std::mutex> lck;
+    };
+
+    template<typename ClockType, typename TaskLockType>
     class Cron;
 
-    template<typename ClockType>
-    std::ostream& operator<<(std::ostream& stream, const Cron<ClockType>& c);
+    template<typename ClockType, typename TaskLockType>
```

Comment:
```suggestion
    template<typename ClockType, typename LockType>
```
LockType says what it does, no need for the extra 'Task'. Need to change this throughout the file too.